<?xml version="1.0" encoding="UTF-8" ?>
<testsuite tests="2" failures="0" name="fatjdbc.ConnectionTest" time="0.676" errors="1" skipped="0">
  <properties>
    <property name="java.runtime.name" value="OpenJDK Runtime Environment"/>
    <property name="java.vm.version" value="14.0.1+14"/>
    <property name="sun.boot.library.path" value="/usr/local/Cellar/openjdk/14.0.1/libexec/openjdk.jdk/Contents/Home/lib"/>
    <property name="maven.multiModuleProjectDirectory" value="/Users/michaelchang/privacy_proxy"/>
    <property name="java.vm.vendor" value="Oracle Corporation"/>
    <property name="java.vendor.url" value="https://openjdk.java.net/"/>
    <property name="guice.disable.misplaced.annotation.check" value="true"/>
    <property name="path.separator" value=":"/>
    <property name="java.vm.name" value="OpenJDK 64-Bit Server VM"/>
    <property name="user.country" value="US"/>
    <property name="sun.java.launcher" value="SUN_STANDARD"/>
    <property name="java.vm.specification.name" value="Java Virtual Machine Specification"/>
    <property name="user.dir" value="/Users/michaelchang/privacy_proxy"/>
    <property name="java.vm.compressedOopsMode" value="Zero based"/>
    <property name="java.runtime.version" value="14.0.1+14"/>
    <property name="os.arch" value="x86_64"/>
    <property name="java.io.tmpdir" value="/var/folders/9y/yz3lkkts2tx99jc422x7cxqr0000gq/T/"/>
    <property name="line.separator" value="
"/>
    <property name="java.vm.specification.vendor" value="Oracle Corporation"/>
    <property name="os.name" value="Mac OS X"/>
    <property name="classworlds.conf" value="/usr/local/Cellar/maven/3.6.3_1/libexec/bin/m2.conf"/>
    <property name="sun.jnu.encoding" value="UTF-8"/>
    <property name="java.library.path" value="/Users/michaelchang/Library/Java/Extensions:/Library/Java/Extensions:/Network/Library/Java/Extensions:/System/Library/Java/Extensions:/usr/lib/java:."/>
    <property name="maven.conf" value="/usr/local/Cellar/maven/3.6.3_1/libexec/conf"/>
    <property name="jdk.debug" value="release"/>
    <property name="java.class.version" value="58.0"/>
    <property name="java.specification.name" value="Java Platform API Specification"/>
    <property name="sun.management.compiler" value="HotSpot 64-Bit Tiered Compilers"/>
    <property name="os.version" value="10.14"/>
    <property name="library.jansi.path" value="/usr/local/Cellar/maven/3.6.3_1/libexec/lib/jansi-native"/>
    <property name="http.nonProxyHosts" value="local|*.local|169.254/16|*.169.254/16"/>
    <property name="user.home" value="/Users/michaelchang"/>
    <property name="user.timezone" value="America/Los_Angeles"/>
    <property name="file.encoding" value="UTF-8"/>
    <property name="java.specification.version" value="14"/>
    <property name="user.name" value="michaelalanchang"/>
    <property name="java.class.path" value="/usr/local/Cellar/maven/3.6.3_1/libexec/boot/plexus-classworlds-2.6.0.jar"/>
    <property name="java.vm.specification.version" value="14"/>
    <property name="sun.arch.data.model" value="64"/>
    <property name="sun.java.command" value="org.codehaus.plexus.classworlds.launcher.Launcher test"/>
    <property name="java.home" value="/usr/local/Cellar/openjdk/14.0.1/libexec/openjdk.jdk/Contents/Home"/>
    <property name="user.language" value="en"/>
    <property name="java.specification.vendor" value="Oracle Corporation"/>
    <property name="java.vm.info" value="mixed mode, sharing"/>
    <property name="java.version" value="14.0.1"/>
    <property name="java.vendor" value="N/A"/>
    <property name="maven.home" value="/usr/local/Cellar/maven/3.6.3_1/libexec"/>
    <property name="file.separator" value="/"/>
    <property name="java.version.date" value="2020-04-14"/>
    <property name="java.vendor.url.bug" value="https://bugreport.java.com/bugreport/"/>
    <property name="sun.io.unicode.encoding" value="UnicodeBig"/>
    <property name="sun.cpu.endian" value="little"/>
    <property name="socksNonProxyHosts" value="local|*.local|169.254/16|*.169.254/16"/>
    <property name="ftp.nonProxyHosts" value="local|*.local|169.254/16|*.169.254/16"/>
  </properties>
  <testcase classname="fatjdbc.ConnectionTest" name="testValid" time="0.676">
    <error message="Error while executing SQL &quot;select * from h2.public.warehouse as wr where wr.w_warehouse_sq_ft &gt; 200&quot;: Schema &quot;H2&quot; not found; SQL statement:
select * from h2.public.warehouse as wr where wr.w_warehouse_sq_ft &gt; 200 [90079-140]" type="java.sql.SQLException">java.sql.SQLException: Error while executing SQL &quot;select * from h2.public.warehouse as wr where wr.w_warehouse_sq_ft &gt; 200&quot;: Schema &quot;H2&quot; not found; SQL statement:
select * from h2.public.warehouse as wr where wr.w_warehouse_sq_ft &gt; 200 [90079-140]
	at org.apache.calcite.avatica.Helper.createException(Helper.java:56)
	at org.apache.calcite.avatica.Helper.createException(Helper.java:41)
	at org.apache.calcite.avatica.AvaticaStatement.executeInternal(AvaticaStatement.java:163)
	at org.apache.calcite.avatica.AvaticaStatement.executeQuery(AvaticaStatement.java:227)
	at fatjdbc.ConnectionTest.testValidImpl(ConnectionTest.java:71)
	at fatjdbc.ConnectionTest.testValid(ConnectionTest.java:94)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.RuntimeException: Schema &quot;H2&quot; not found; SQL statement:
select * from h2.public.warehouse as wr where wr.w_warehouse_sq_ft &gt; 200 [90079-140]
	at fatjdbc.PrivacyMetaImpl.propagate(PrivacyMetaImpl.java:1050)
	at fatjdbc.PrivacyMetaImpl.prepareAndExecute(PrivacyMetaImpl.java:420)
	at org.apache.calcite.avatica.AvaticaConnection.prepareAndExecuteInternal(AvaticaConnection.java:675)
	at org.apache.calcite.avatica.AvaticaStatement.executeInternal(AvaticaStatement.java:156)
	... 33 more
</error>
    <system-out>Value = file:/Users/michaelchang/privacy_proxy/target/test-classes/tpcds.sql
Value = file:/Users/michaelchang/privacy_proxy/target/test-classes/tpcds_cubes.sql
Value = file:/Users/michaelchang/privacy_proxy/target/test-classes/tpcds_views.sql
establishing connection
inside privacy meta impl
connection established
Creating statement in privacymetimpl
fatjdbc.PrivacyConnectionImpl@2a62b5bc
statement created
in prepareAndExecute2 in privacymetaimpl
inside callback
cleared callback
inside constructor
inside query context
inside schema factory class
inside privacy factory
finished parsing
adding root schema
trying to initialize
class plugin.jdbc.H2Db
get schemas
getting connections
Creating statement
executing query
getting schema from result set query
finished building
iterating through rs
PUBLIC
about to finish building
get schema fromm result set
Trying to initialize2initializing root schema
data source schema
in query context
Parsing in privacyparser
parsing sql
Executing inside Plan Executor
Executing queries inside PrivacyQueryExecutor
</system-out>
    <system-err>log4j:WARN No appenders could be found for logger (plugin.jdbc.JdbcDB).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
</system-err>
  </testcase>
  <testcase classname="fatjdbc.ConnectionTest" name="testEmptyDataSources" time="0"/>
</testsuite>